### DISCUSSION TITLE
Interpolation of public variables with numeric values inside kustomize patches

### DISCUSSION THREAD

**Discussion by elmalakomar** [mia-platform/community#492] [Q&A]
### What product are you having troubles with?

Console

### What Console version are you using?

v13.0.0

### Description

Hi,
I'm trying to define a kustomize patch file that will create a sidecar for a container.
One of the required environment variables is interpolating a numeric value (like 8808) from a Public Variable (lets call it MY_PORT) defined in the Design section of the Console.
 
 So in my `sidecar.patch.yaml` file I have this definition of environment variable:
 ```
 env:
  - name: MY_PORT
    value: {{MY_PORT}}
 ```

This configuration when trying to deploy is returning this error:
```
fails to convert unstructured pod pod-name spec: unrecognized type: string
fails to insert dependencies: unrecognized type: string
fails to prepare resources: unrecognized type: string
```

This behaviour is happening only when I have a numeric value. How can I deal with this problem? 
Thanks

### Actual Outcome

Failed deploy with error:
```
fails to convert unstructured pod pod-name spec: unrecognized type: string
fails to insert dependencies: unrecognized type: string
fails to prepare resources: unrecognized type: string
```

### Expected Outcome

Success deploy and environment with correct variable

**Comment by fredmaggiowski** [comment]
Ho @elmalakomar, have you tried putting the interpolation string between double quotes? Yaml should implicitly convert the number to string, but you never know 🤔
